packages/next/src/build/webpack-config.ts

```diff
@@ -1199,13 +1187,28 @@ export default async function getBaseWebpackConfig(
       ...(config.experimental.flyingShuttle
         ? {
             // ensure we only use contenthash as it's more deterministic
-            filename: isNodeOrEdgeCompilation
-              ? dev || isEdgeServer
-                ? `[name].js`
-                : `../[name].js`
-              : `static/chunks/${isDevFallback ? 'fallback/' : ''}[name]${
-                  dev ? '' : '-[contenthash]'
-                }.js`,
+            filename: (p) => {
+              if (isNodeOrEdgeCompilation) {
```

Comment:
should the changes here be gated?

Reply:
This is already gated by the `isFlyingShuttle` flag above